assets/app/scripts/directives/nav.js

```diff
@@ -58,10 +58,11 @@ angular.module('openshiftConsole')
 
           sortedProjects = $filter('orderByDisplayName')(projects);
           options = _.map(sortedProjects, function(item) {
+            // truncates @ 35 as this prevents words from clobbering the caret
             return $('<option>')
-                      .attr("value", item.metadata.name)
+                      .attr('value', $filter('truncate')(item.metadata.name, 35))
```

Comment:
not sure how this would work, this is setting the value on the option tag to something thats not a valid project name, its going to cause you to navigate to a project that doesnt exist.

as a side note, when i have a long project name i am seeing it truncating with the current CSS, what exactly are you seeing.